src/lib/icons.tsx

```diff
@@ -37,8 +39,13 @@ async function getIcons(dirName: string): Promise<Icon[]> {
   const fileNames = await fs.readdir(dirPath);
   return await Promise.all(
     fileNames.map(async (fileName) => {
+      const id = fileName.replace(/\.[^/.]+$/, '');
+
+      const md = metaData.find((m) => m.id === id);
+
       return {
-        title: fileName.replace(/\.[^/.]+$/, ''),
+        title: md && md.title ? md.title : fileName.replace(/\.[^/.]+$/, ''),
+        tags: md && md.tags ? md.tags : [],
```

Comment:
The same here we can transform it to  `tags: md?.tags || [],` to make it a bit easier to read